docs/en/latest/plugins/hmac-auth.md

```diff
@@ -63,312 +69,474 @@ admin_key=$(yq '.deployment.admin.admin_key[0].key' conf/config.yaml | sed 's/"/
-
-- **HTTP Method** : HTTP request method in uppercase. For example, GET, PUT, POST etc.
-- **HTTP URI** : HTTP URI. Should start with "/" and "/" denotes an empty path.
-- **Date** : Date in the HTTP header in GMT format.
-- **canonical_query_string** : The result of encoding the query string in the URL (the string "key1 = value1 & key2 = value2" after the "?" in the URL).
-- **signed_headers_string** : Concatenation of the specified request headers.
+```python title="hmac-sig-header-gen.py"
+import hmac
+import hashlib
+import base64
+from datetime import datetime, timezone
+
+key_id = "john-key"                # key id
+secret_key = b"john-secret-key"    # secret key
+request_method = "GET"             # HTTP method
+request_path = "/get"              # route URI
+algorithm= "hmac-sha256"           # can use other algorithms in allowed_algorithms
+
+# get current datetime in GMT
+# note: the signature will become invalid after the clock skew (default 300s)
+# you can regenerate the signature after it becomes invalid, or increase the clock
+# skew to prolong the validity within the advised security boundary
+gmt_time = datetime.now(timezone.utc).strftime('%a, %d %b %Y %H:%M:%S GMT')
+
+# construct the signing string (ordered)
+# the date and any subsequent custom headers should be lowercased and separated by a
+# single space character, i.e. `<key>:<space><value>`
+# https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-12#section-2.1.6
+signing_string = (
+  f"{key_id}\n"
+  f"{request_method} {request_path}\n"
+  f"date: {gmt_time}\n"
+)
+
+# create signature
+signature = hmac.new(secret_key, signing_string.encode('utf-8'), hashlib.sha256).digest()
+signature_base64 = base64.b64encode(signature).decode('utf-8')
+
+# construct the request headers
+headers = {
+  "Date": gmt_time,
+  "Authorization": (
+    f'Signature keyId="{key_id}",algorithm="{algorithm}",'
+    f'headers="@request-target date",'
+    f'signature="{signature_base64}"'
+  )
+}
+
+# print headers
+print(headers)
+```
 
-:::tip
+Run the script:
 
-If any of the terms are missing, they are replaced by an empty string.
+```shell
+python3 hmac-sig-header-gen.py
+```
 
-:::
+You should see the request headers printed:
 
-**The algorithm for generating `canonical_query_string` is described below:**
-
-1. Extract the query terms from the URL.
-2. Split the query terms into key-value pairs by using `&` as the separator.
-3. If `encode_uri_params` is `true`:
-   1. If there are only keys, the conversion formula is `uri_encode(key) + "="`.
-   2. If there are both keys and values, the conversion formula is `uri_encode(key) + "=" + uri_encode(value)`. Here, the value can even be an empty string.
```

Comment:
Is this algorithm documented anywhere else or is it no longer needed?

Reply:
@Revolyssup hello could you help evaluate if the above about constructing `canonical_query_string` is still relevant? I can't find anything about "query string" in the RFC.